### PR TITLE
Support evidence type in api/search

### PR DESCRIPTION
**Pull request by zhx828**
Not all indicator calls may require all information queried especially when only oncogenic is needed, all therapeutics info any be ignored

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java [hunk 1]
```diff
@@ -170,60 +187,64 @@ public static IndicatorQueryResp processQuery(Query query, String geneStatus,
...
+                        indicatorQuery.setOncogenic(oncogenicity.getOncogenic());
+                    }
                 }
 
-                // Only set oncogenicity if no previous data assigned.
-                if (indicatorQuery.getOncogenic() == null && oncogenicity != null) {
-                    indicatorQuery.setOncogenic(oncogenicity.getOncogenic());
+                if (hasTreatmentEvidence) {
+                    treatmentEvidences = EvidenceUtils.keepHighestLevelForSameTreatments(
+                        EvidenceUtils.getRelevantEvidences(query, source, geneStatus,
```
Comment (JiaoJiao123) [inline core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java]: Should we do the null check on the getRelevantEvidences() return before calling keepHighestLevelForSameTreatments()?

File: core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java [hunk 2]
```diff
@@ -234,17 +255,19 @@ public static IndicatorQueryResp processQuery(Query query, String geneStatus,
...
-                        EvidenceUtils.convertEvidenceLevel(
-                            EvidenceUtils.getEvidence(Collections.singletonList(oncogenicMutation),
-                                MainUtils.getTreatmentEvidenceTypes(),
-                                (levels != null ?
-                                    new HashSet<>(CollectionUtils.intersection(levels,
-                                        LevelUtils.getPublicAndOtherIndicationLevels())) : LevelUtils.getPublicAndOtherIndicationLevels())), new HashSet<>(oncoTreeTypes))));
+                    if (hasTreatmentEvidence) {
+                        treatmentEvidences.addAll(EvidenceUtils.keepHighestLevelForSameTreatments(
+                            EvidenceUtils.convertEvidenceLevel(
+                                EvidenceUtils.getEvidence(Collections.singletonList(oncogenicMutation),
```
Comment (JiaoJiao123) [inline core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java]: Same questions here. Should we do the null check before passing the return as the parameter for the next round call?

File: core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java (hunk 1 above)
Comment (zhx828) [reply core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java]: getRelevantEvidences will always return an empty list if nothing has been found.

File: core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java (hunk 2 above)
Comment (zhx828) [reply core/src/main/java/org/mskcc/cbio/oncokb/util/IndicatorUtils.java]: same here, an empty list will always be returned.

**Pull request merged**